### PR TITLE
Fix style input value

### PR DESCRIPTION
**Pull request by evulins**
Allows displaying tag in the input field.

### CODE REVIEW

File: src/components/atoms/Tag/index.tsx [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
+  text,
+  fontSize,
+  onRemoveTag,
+  isDroppable = false,
+}) => (
+  <S.TagWrapper fontSize={fontSize}>
+    {text}
+    {isDroppable && (
+      <S.RemoveButton
+        aria-label="Usuń tag"
```
Comment (franiakat) [inline src/components/atoms/Tag/index.tsx:16]: 👏

File: src/components/atoms/Tag/interface.ts [hunk 2]
```diff
@@ -1,9 +1,19 @@
...
+};
+
+type DroppableTag = {
+  text: string;
+  isDroppable?: true;
+  fontSize?: string;
+  onRemoveTag?: () => void;
+};
+
+export type TagProps = Tag | DroppableTag;
```
Comment (franiakat) [inline src/components/atoms/Tag/interface.ts]:
```
type CommonTagProps = {
  text: string;
  fontSize?: string;
};

type Tag = {
  isDroppable?: false;
  onRemoveTag?: never;
};

type DroppableTag = {
  isDroppable?: true;
  onRemoveTag?: () => void;
};

export type TagProps = (Tag | DroppableTag) & CommonTagProps;
```

File: src/components/molecules/SearchAutocomplete/index.test.tsx [hunk 3]
```diff
@@ -1,5 +1,6 @@
 import { screen, fireEvent, within } from "@testing-library/react";
 import userEvent from "@testing-library/user-event";
+import { debug } from "util";
```
Comment (franiakat) [inline src/components/molecules/SearchAutocomplete/index.test.tsx]: leftover ;)

File: src/components/molecules/SearchAutocomplete/index.test.tsx [hunk 4]
```diff
@@ -45,16 +46,16 @@ describe("Input component", () => {
     expect(input.value).toBe("");
     expect(screen.queryByRole("listbox")).not.toBeInTheDocument();
 
-    fireEvent.change(input, { target: { value: "Oce" } });
-    expect(input).toHaveValue("Oce");
+    fireEvent.change(input, { target: { value: "Ciągi" } });
```
Comment (franiakat) [inline src/components/molecules/SearchAutocomplete/index.test.tsx]: You can also use [userEvent lib](https://testing-library.com/docs/ecosystem-user-event/#typeelement-text-options), it's not a crazy event, but the syntax is more concise ;)

File: src/components/molecules/SearchAutocomplete/index.test.tsx [hunk 5]
```diff
@@ -120,4 +121,51 @@ describe("Input component", () => {
...
+  it("should display input with tag as a value and return to default input after removing tag", async () => {
+    testRenderer(<SearchAutocomplete setTag={setTag} />);
+    let input = (await screen.findByLabelText(
+      "Szukaj problemów"
+    )) as HTMLInputElement;
+
+    fireEvent.change(input, { target: { value: "J" } });
+    expect(screen.queryByText("Jedzenie")).toBeInTheDocument();
+    fireEvent.keyDown(input, { key: "ArrowDown", code: 40 });
+    fireEvent.keyDown(input, { key: "Enter", code: 13 });
```
Comment (franiakat) [inline src/components/molecules/SearchAutocomplete/index.test.tsx]: I think the code can be skipped here.

File: src/components/molecules/SearchAutocomplete/index.test.tsx [hunk 6]
```diff
@@ -120,4 +121,51 @@ describe("Input component", () => {
...
+
+    const removeTagButton = (await screen.findByRole("button", {
+      name: /Usuń tag/i,
+    })) as HTMLButtonElement;
+    expect(removeTagButton).toBeInTheDocument();
+    fireEvent.click(removeTagButton);
+
+    let defaultInput = (await screen.findByLabelText(
```
Comment (franiakat) [inline src/components/molecules/SearchAutocomplete/index.test.tsx]: You can use `const` here.

File: src/components/molecules/SearchAutocomplete/index.tsx [hunk 7]
```diff
@@ -18,6 +19,7 @@ const SearchAutocomplete: FC<SearchAutocompleteProps> = ({ setTag }) => {
   const [showSuggestions, setShowSuggestions] = useState(false);
   const [searchTerm, setSearchTerm] = useState("");
   const [showClearButton, setShowClearButton] = useState(false);
+  const [suggestionType, setSuggestionType] = useState("");
```
Comment (franiakat) [inline src/components/molecules/SearchAutocomplete/index.tsx]: I think we don't need additional state here (`[suggestionType, setSuggestionType]`), I'd set tag only when suggestion id of type TAG and therefore whenever we check whether tag exists we can assume the suggestion is of type TAG -> we use our fake input with proper styling. I have checked it locally and works good :)

File: src/components/molecules/SearchAutocomplete/index.tsx [hunk 8]
```diff
@@ -106,34 +112,47 @@ const SearchAutocomplete: FC<SearchAutocompleteProps> = ({ setTag }) => {
...
-          value={searchTerm}
-          onChange={handleChange}
-          onKeyDown={handleKeyDown}
-          onKeyUp={handleKeyUp}
-          onSubmit={(e) => {
-            e.preventDefault();
-          }}
-          showSuggestions={showSuggestions}
-        />
+        {searchTerm && suggestionType === "tag" ? (
```
Comment (franiakat) [inline src/components/molecules/SearchAutocomplete/index.tsx]: Here we'd have 
```suggestion
        {searchTerm && !!tag ? (
```
and `tag` would come from a parent altogether with `setTag`

File: src/components/molecules/SearchAutocomplete/index.tsx [hunk 9]
```diff
@@ -106,34 +112,47 @@ const SearchAutocomplete: FC<SearchAutocompleteProps> = ({ setTag }) => {
...
+          <>
+            <S.Button aria-label="Remove button" onClick={() => clearInput()}>
+              <span id="Remove button" hidden>
+                Remove input text button
+              </span>
 
-            <S.RemoveIcon aria-hidden="true" />
-          </S.Button>
+              <S.RemoveIcon aria-hidden="true" />
+            </S.Button>
```
Comment (franiakat) [inline src/components/molecules/SearchAutocomplete/index.tsx]: hm, I think I wouldn't keep it for tag inputs ;)

File: src/components/molecules/SearchAutocomplete/inputWithTags.tsx [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+  onRemoveTag,
+  setSuggestionType,
+}) => {
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLSpanElement>) => {
+    if (e.key === "Backspace") {
+      setSuggestionType("");
+    }
+    if (e.key === "Enter") {
+      e.preventDefault();
+      return false;
```
Comment (franiakat) [inline src/components/molecules/SearchAutocomplete/inputWithTags.tsx]: we could use switch case here ;)

File: src/components/molecules/SearchAutocomplete/interface.ts [hunk 11]
```diff
@@ -19,6 +19,16 @@ export interface InputStyleProps {
   showSuggestions: boolean;
 }
 
+export interface InputWithTagsProps {
+  onRemoveTag: () => void;
+  text: string;
+  setSuggestionType: (type: string) => void;
```
Comment (franiakat) [inline src/components/molecules/SearchAutocomplete/interface.ts]: I'd keep the functions at the end, but it's just a nitpick and a matter of style. ;)

File: src/components/molecules/SearchAutocomplete/styles.tsx [hunk 12]
```diff
@@ -63,15 +57,29 @@ const Input = styled.input<InputStyleProps>`
...
+  border-radius: ${(props) => (props.showSuggestions ? 0 : "5px")};
+  border-top-left-radius: 5px;
+  border-top-right-radius: 5px;
+  ::placeholder {
+    color: ${placeholder};
+  }
+  &:focus-visible {
+    outline: none;
+  }
+  ${inputStyles}
```
Comment (franiakat) [inline src/components/molecules/SearchAutocomplete/styles.tsx:70]: nice!

**Review by franiakat — CHANGES_REQUESTED** [review]

File: src/components/molecules/SearchAutocomplete/index.test.tsx (hunk 5 above)
Comment (evulins) [reply src/components/molecules/SearchAutocomplete/index.test.tsx]: How can I trigger to show input with tags then?

File: src/components/molecules/SearchAutocomplete/index.test.tsx [hunk 13]
```diff
@@ -120,4 +122,51 @@ describe("Input component", () => {
...
+
+    fireEvent.change(input, { target: { value: "J" } });
+    expect(screen.queryByText("Jedzenie")).toBeInTheDocument();
+    fireEvent.keyDown(input, { key: "ArrowDown", code: 40 });
+    fireEvent.keyDown(input, { key: "Enter", code: 13 });
+
+    const removeTagButton = await screen.findByRole("button", {
+      name: /Usuń tag/i,
+    });
+    expect(removeTagButton).toBeInTheDocument();
```
Comment (bazelajoanna) [inline src/components/molecules/SearchAutocomplete/index.test.tsx]: Do we need to check this in this test case?

Comment (franiakat) [reply src/components/molecules/SearchAutocomplete/index.test.tsx]: No, we can remove it on the branch with tests fixing.

File: src/components/molecules/SearchAutocomplete/index.test.tsx (hunk 6 above)
Comment (franiakat) [reply src/components/molecules/SearchAutocomplete/index.test.tsx]: We can also remove this on a branch with tests. :)

**Review by franiakat — APPROVED** [review]

**Comment by franiakat** [comment]
sqash

**Pull request merged**